lib/SILOptimizer/Utils/Differentiation/PullbackEmitter.cpp

```diff
@@ -954,93 +1041,42 @@ AllocStackInst *PullbackEmitter::getArrayAdjointElementBuffer(
   auto *eltIndexInt = builder.createStruct(loc, intType, {eltIndexLiteral});
   auto *swiftModule = getModule().getSwiftModule();
   auto *diffProto = ctx.getProtocol(KnownProtocolKind::Differentiable);
-  auto diffConf = swiftModule->lookupConformance(eltTanAstType, diffProto);
+  auto diffConf = swiftModule->lookupConformance(eltTanType, diffProto);
   assert(!diffConf.isInvalid() && "Missing conformance to `Differentiable`");
   auto *addArithProto = ctx.getProtocol(KnownProtocolKind::AdditiveArithmetic);
-  auto addArithConf =
-      swiftModule->lookupConformance(eltTanAstType, addArithProto);
+  auto addArithConf = swiftModule->lookupConformance(eltTanType, addArithProto);
   assert(!addArithConf.isInvalid() &&
          "Missing conformance to `AdditiveArithmetic`");
-  auto subMap = SubstitutionMap::get(subscriptFnGenSig, {eltTanAstType},
+  auto subMap = SubstitutionMap::get(subscriptFnGenSig, {eltTanType},
                                      {addArithConf, diffConf});
   // %elt_adj = alloc_stack $T.TangentVector
-  auto *eltAdjBuffer = builder.createAllocStack(loc, eltTanType);
+  // Create a local allocation and register it.
+  localAllocBuilder.setInsertionPoint(
+      getPullback().getEntryBlock(),
+      getNextFunctionLocalAllocationInsertionPoint());
+  auto *eltAdjBuffer = localAllocBuilder.createAllocStack(loc, eltTanSILType);
+  functionLocalAllocations.push_back(eltAdjBuffer);
+  // Temporarily change global builder insertion point and emit zero into the
```

Comment:
Here are the errors from commenting the highlighted code (zero initialization + destroy): https://gist.github.com/dan-zheng/56e1233db1646cd4aceaeef23d02555b
```
SIL memory lifetime failure in @AD__$s4mainyycfU6_24controlFlowNestedLiteralL_ySay23DifferentiationUnittest7TrackedVySfGGAF_AFSbtF__pullback_src_0_wrt_0_1: memory is not initialized, but should
memory location:   %27 = alloc_stack $Array<Tracked<Float>>.DifferentiableView // users: %1616, %1615, %536
at instruction:   %536 = apply %534<Array<Tracked<Float>>.DifferentiableView>(%152, %27, %535) : $@convention(witness_method: AdditiveArithmetic) <τ_0_0 where τ_0_0 : AdditiveArithmetic> (@inout τ_0_0, @in_guaranteed τ_0_0, @thick τ_0_0.Type) -> ()
...
```

This error occurs for functions with control flow, since the local allocation isn't guaranteed to be initialized along all control flow paths.